lib/scraper.py

```diff
@@ -1,5 +1,40 @@
-from utils import Stock
+from typing import List
+import requests
+import re
+
+from bs4 import BeautifulSoup
+from lib.utils.classes import ChronologicalFloat, Stock
+from lib.utils.args import debug
+from bs4 import BeautifulSoup, Tag
 
 # Given a ticker, scrapes the web for information and returns a Stock object
-def scrape(ticker: str) -> Stock:
-    raise Exception("scraper.scrape: Not implemented")
+# Ticker: pd.DataFrame: Name, Ticker, URL
+def scrape(session: requests.Session, ticker: str) -> Stock:
+    debug("Scraping %s" % ticker)
+    url = "https://www.screener.in/company/%s/consolidated/" % ticker
+    page = session.get(url)
+    soup = BeautifulSoup(page.text, 'html.parser')
+    name = soup.find('h1').text
+    [net_profit] = parse_table(soup.select_one('#profit-loss .data-table'), ['net profit'])
```

Comment:
The parse data is in the form of a list of dictionaries. Can we directly add these values into a dataframe with the year & financial metrics being columns? Plus can we add range for years?

Reply:
I think that makes sense - let me do that

Reply:
Let me spend a few hours learning dataframes